pinc/ProjectTransition.inc

```diff
@@ -294,21 +293,16 @@ class ProjectTransition
         if (!is_null($pool) && $this->to_state == $pool->project_checkedout_state) {
             // we want the project event to record who it's been
             // checked out to
-            $project = new Project($projectid);
             $checkedoutby = $project->checkedoutby;
         }
         // log the project event
-        $err = log_project_event(
-            $projectid,
+        $project->log_project_event(
             $who,
             'transition',
             $this->from_state,
             $this->to_state,
             ($checkedoutby != '' ? "out_to: $checkedoutby" : @$extras['details'])
```

Comment:
Should we do something better instead of `@extras`? extras is alawys an empty array so far as I can tell so prehaps remove it.

Reply:
Very often, but not quite always. `$extras` is used within `tools/project_manager/autorelease.inc` to include the queue name:
```
    $error_msg = project_transition($projectid, $new_state, PT_AUTO, ['details' => "via_q: $queue_name"]);
calls:
    $error = $transition->do_state_change($project, $who, $extras);
calls:
    $project->log_project_event(...)
```
I'm not 100% certain that it's actually used in this path, but it could be.